src/main/java/ch/wisv/payments/rest/eventsync/EventsSyncRestController.java

```diff
@@ -0,0 +1,111 @@
+package ch.wisv.payments.rest.eventsync;
+
+import ch.wisv.payments.admin.products.ProductService;
+import ch.wisv.payments.exception.ProductNotFoundException;
+import ch.wisv.payments.model.eventsync.ProductEventsSync;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.validation.annotation.Validated;
+import org.springframework.web.bind.annotation.*;
+
+import javax.servlet.http.HttpServletRequest;
+import javax.validation.constraints.NotNull;
+import java.nio.charset.Charset;
+import java.util.Base64;
+
+import static ch.wisv.payments.util.ResponseEntityBuilder.createResponseEntity;
+
+@RestController
+@CrossOrigin
+@RequestMapping(value = "/api/chevents/sync")
+@Validated
+public class EventsSyncRestController {
+
+    @Value("${events.username}")
+    @NotNull
+    private String username;
+
+    @Value("${events.password}")
+    @NotNull
+    private String password;
+
+    /**
+     * ProductService.
+     */
+    private final ProductService productService;
+
+    private final EventsSyncProductService eventsSyncProductService;
+
+    /**
+     * Constructor ProductService.
+     *
+     * @param productService of type ProductService
+     * @param eventsSyncProductService of type EventsSyncProductService
+     */
+    public EventsSyncRestController(ProductService productService, EventsSyncProductService eventsSyncProductService) {
+        this.productService = productService;
+        this.eventsSyncProductService = eventsSyncProductService;
+    }
+
+
+    @PostMapping("/product/")
+    public ResponseEntity<?> eventSync(HttpServletRequest request, @RequestBody ProductEventsSync productEventsSync) {
+        String[] credentials = this.decryptBasicAuthHeader(request.getHeader("Authorization"));
+
+        if (!this.authChEvents(credentials)) {
+            return createResponseEntity(HttpStatus.UNAUTHORIZED, "User is not authorized", null);
+        }
+
+        switch (productEventsSync.getTrigger()) {
+            case "PRODUCT_CREATE_EDIT":
+                this.determineCreateOrUpdate(productEventsSync);
+                break;
+            case "PRODUCT_DELETE":
+                eventsSyncProductService.deleteProduct(productEventsSync);
+                break;
+            default:
+                return createResponseEntity(HttpStatus.BAD_REQUEST, "Events trigger not supported!", null);
+        }
+
+        return createResponseEntity(HttpStatus.OK, null, null);
```

Comment:
This should be more specific, with HttpStatus.CREATED when a new product is created

Reply:
This should be a HttpStatus.OK, otherwise CH events thinks the sync failed.